src/pickup.c

```diff
@@ -2443,15 +2452,30 @@ exchange_objects_with_mon(struct monst *mtmp, boolean taking)
                     /* extra delay for removing a cloak */
                     m_delay += 2;
                 }
-                pline("%s %s %s %s.", Monnam(mtmp),
-                      m_delay > 1 ? "begins removing" : "removes",
-                      mhis(mtmp), xname(otmp));
+                if ((unwornmask & W_SADDLE) != 0) {
+                    You("remove %s from %s.", the(xname(otmp)),
+                        x_monnam(mtmp, ARTICLE_THE, (char *) 0,
+                                 SUPPRESS_SADDLE, FALSE));
+                    /* unstrapping a saddle takes additional time */
+                    time_taken += rn2(3);
+                }
+                else {
+                    pline("%s %s %s %s.", Monnam(mtmp),
+                          m_delay > 1 ? "begins removing" : "removes",
+                          mhis(mtmp), xname(otmp));
+                }
                 mtmp->mfrozen = m_delay;
                 /* unwear the item now */
                 update_mon_intrinsics(mtmp, otmp, FALSE, FALSE);
-                otmp->owornmask = 0;
                 if (mtmp->mfrozen) { /* might be 0 */
                     mtmp->mcanmove = 0;
+                    otmp->owornmask = 0L;
+                    /* normally extract_from_minvent handles this stuff, but
+                     * since we are setting owornmask to 0 now we have to
+                     * do it here. */
+                    otmp->owt = weight(otmp); /* reset armor weight */
+                    mtmp->misc_worn_check &= ~unwornmask;
+                    check_gear_next_turn(mtmp);
```

Comment:
I think this should specifically not be called here - otherwise when they unfreeze, they're likely to just put it back on again, no?

Reply:
Yeah, you're probably right.  When I tested it I didn't see the monsters put armor back on, so I thought you may have marked recently removed armor in some way to discourage that behavior.  If that's not the case I guess it's best to get rid of it.